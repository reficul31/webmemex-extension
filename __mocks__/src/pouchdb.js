import PouchDB from 'pouchdb-node'
import PouchDBMemory from 'pouchdb-adapter-memory'

PouchDB.plugin(PouchDBMemory)

const pouchdbOptions = {
    name: 'testdb',
    auto_compaction: true,
    adapter: 'memory',
}

const db = PouchDB(pouchdbOptions)
export default db
