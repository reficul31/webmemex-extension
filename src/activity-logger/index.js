// Stuff that is to be accessible from other modules (folders)

import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const visitKeyPrefix = 'visit/'

/**
* Creates an _id string given the variables, or vice versa parses such strings
* We simply use the creation time for the id, for easy chronological sorting.
* We add a random string we call a 'nonce' to prevent accidental collisions.
*
* @param {Number} timestamp - timestamp of the document
* @param {Number} nonce - random 10 digit number
*
* @returns {String} - string id of the document
*/
export const convertVisitDocId = docuri.route(`${visitKeyPrefix}:timestamp/:nonce`)

/**
* Extract the type, timestamp and nonce from an ID of a document
*
* @param {String} - string id of the document
*/
const convertAnyTimestampedDocId = docuri.route(':type/:timestamp/:nonce')
export const getTimestamp = doc =>
    Number.parseInt(convertAnyTimestampedDocId(doc._id).timestamp)

/**
* Generate a visit document ID from the timestamp and a random string
*
* @param {Number} timestamp - timestamp of the visit document
* @param {Number} nonce - random 10 digit number
*
* @returns {string} - document id of the visit document
*/
export function generateVisitDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertVisitDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}

/**
* Tell whether a page can be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page is loggable
*/
export function isLoggable({url}) {
    // Only http(s) pages. Ignoring data URLs, newtab, ...
    const loggableUrlPattern = /^https?:\/\//
    return loggableUrlPattern.test(url)
}

/**
* Tell whether a page should be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page should be stored
*/
export function shouldBeLogged({url}) {
    // Currently, we log everything that we think we can log.
    // TODO Add ignore-/blacklist lookup here (issue #22, #94)
    return isLoggable({url})
}
