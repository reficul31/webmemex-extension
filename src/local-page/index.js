import get from 'lodash/fp/get'

/**
* Extract the local page URL from the page instance.
*
* @param {Page} page - stored page in the database
*
* @returns {string} - URL used to refer to the local page
*/
export function hrefForLocalPage({page}) {
    // Check if it has a stored page attached at all.
    if (!get(['_attachments', 'frozen-page.html'])(page)) {
        return undefined
    }
    const pageId = page._id
    const url = `/local-page.html?page=${encodeURIComponent(pageId)}`
    const hash = (page.url && page.url.split('#')[1])
    const href = (hash !== undefined) ? url + '#' + hash : url
    return href
}
