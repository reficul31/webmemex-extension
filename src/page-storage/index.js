import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const pageKeyPrefix = 'page/'

export const convertPageDocId = docuri.route(`${pageKeyPrefix}:timestamp/:nonce`)

/**
* Generate a page doc id
*
* @param {Number} timestamp - timestamp of the page
* @param {string} nonce - random string of 10 digit number
*
* @returns {string} - doc id of the page as a string
*/
export function generatePageDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertPageDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}
