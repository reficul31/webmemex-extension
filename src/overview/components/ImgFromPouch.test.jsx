/* eslint-env jest */
/* eslint import/namespace: "off" */

import React from 'react'
import ImgFromPouch from 'src/overview/components/ImgFromPouch'
import { shallow } from 'enzyme'
import * as db from 'src/pouchdb'
import searchData from 'src/overview/components/result.data.json'

describe('ImgFromPouch', () => {
    const doc = searchData.rows[0].doc
    const dataUri = 'data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t/f3//Ub//ge8WSLf/rhf/3kdbW1mxsbP//mf///yH5BAAAAAAALAAAAAAQAA4AAARe8L1Ekyky67QZ1hLnjM5UUde0ECwLJoExKcppV0aCcGCmTIHEIUEqjgaORCMxIC6e0CcguWw6aFjsVMkkIr7g77ZKPJjPZqIyd7sJAgVGoEGv2xsBxqNgYPj/gAwXEQA7'

    test('should test against the given snapshot', async () => {
        db.getAttachmentAsDataUrl = jest.fn().mockReturnValue(new Promise((resolve, reject) => {
            resolve(dataUri)
        }))
        const tree = await shallow(
            <ImgFromPouch className='favicon' doc={doc.page} attachmentId='favIcon' />
        )
        expect(tree).toMatchSnapshot()
    })

    test('should convert blob to data uri', async () => {
        db.getAttachmentAsDataUri = jest.fn().mockReturnValue(new Promise((resolve, reject) => {
            resolve(dataUri)
        }))
        const doc = searchData.rows[0].doc
        await shallow(
            <ImgFromPouch className='favicon' doc={doc.page} attachmentId='favIcon' />
        )
        expect(db.getAttachmentAsDataUrl).toBeCalled()
    })
})
