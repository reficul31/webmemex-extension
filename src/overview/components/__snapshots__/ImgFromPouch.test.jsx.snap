// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgFromPouch should test against the given snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <img
    className="favicon"
    src={undefined}
/>,
  "nodes": Array [
    <img
      className="favicon"
      src={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ImgFromPouch
        attachmentId="favIcon"
        className="favicon"
        doc={
                Object {
                        "_attachments": Object {
                          "favIcon": Object {
                            "content_type": "image/x-icon",
                            "digest": "md5-80GKRD59hBCXxxTWnsS8uA==",
                            "length": 5430,
                            "revpos": 2,
                            "stub": true,
                          },
                          "frozen-page.html": Object {
                            "content_type": "text/html;charset=utf-8",
                            "digest": "md5-rL1HMBSNfY37snk8/ZBX1w==",
                            "length": 549162,
                            "revpos": 6,
                            "stub": true,
                          },
                          "screenshot": Object {
                            "content_type": "image/png",
                            "digest": "md5-iJNPepFWRMhtmqAk/y5eAA==",
                            "length": 41222,
                            "revpos": 5,
                            "stub": true,
                          },
                        },
                        "_id": "page/1499620431844/1337776782",
                        "_rev": "6-3b60ef0bcdebea7c54615efb445613a3",
                        "extractedMetadata": Object {
                          "icon_url": "https://www.google.co.in/images/branding/product/ico/googleg_lodp.ico",
                          "provider": "google",
                          "title": "Google",
                          "url": "https://www.google.co.in/?ei=R2RiWZz1I4OW8QeqwZTIDA",
                        },
                        "extractedText": Object {
                          "bodyInnerText": "Screen-reader users, click here to turn off Google Instant.
                Gmail
                Images
                Sign in
                
                
                India
                Google.co.in offered in: हिन्दी বাংলা తెలుగు मराठी தமிழ் ગુજરાતી ಕನ್ನಡ മലയാളം ਪੰਜਾਬੀ
                Privacy Terms Settings  Use Google.com Advertising Business About
                ",
                        },
                        "title": "",
                        "url": "https://www.google.co.in/?gfe_rd=cr&ei=R2RiWZz1I4OW8QeqwZTIDA",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ImgFromPouch {
        "_isMounted": true,
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "attachmentId": "favIcon",
          "className": "favicon",
          "doc": Object {
            "_attachments": Object {
              "favIcon": Object {
                "content_type": "image/x-icon",
                "digest": "md5-80GKRD59hBCXxxTWnsS8uA==",
                "length": 5430,
                "revpos": 2,
                "stub": true,
              },
              "frozen-page.html": Object {
                "content_type": "text/html;charset=utf-8",
                "digest": "md5-rL1HMBSNfY37snk8/ZBX1w==",
                "length": 549162,
                "revpos": 6,
                "stub": true,
              },
              "screenshot": Object {
                "content_type": "image/png",
                "digest": "md5-iJNPepFWRMhtmqAk/y5eAA==",
                "length": 41222,
                "revpos": 5,
                "stub": true,
              },
            },
            "_id": "page/1499620431844/1337776782",
            "_rev": "6-3b60ef0bcdebea7c54615efb445613a3",
            "extractedMetadata": Object {
              "icon_url": "https://www.google.co.in/images/branding/product/ico/googleg_lodp.ico",
              "provider": "google",
              "title": "Google",
              "url": "https://www.google.co.in/?ei=R2RiWZz1I4OW8QeqwZTIDA",
            },
            "extractedText": Object {
              "bodyInnerText": "Screen-reader users, click here to turn off Google Instant.
Gmail
Images
Sign in


India
Google.co.in offered in: हिन्दी বাংলা తెలుగు मराठी தமிழ் ગુજરાતી ಕನ್ನಡ മലയാളം ਪੰਜਾਬੀ
Privacy Terms Settings  Use Google.com Advertising Business About
",
            },
            "title": "",
            "url": "https://www.google.co.in/?gfe_rd=cr&ei=R2RiWZz1I4OW8QeqwZTIDA",
          },
        },
        "refs": Object {},
        "state": Object {
          "dataUri": "data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t/f3//Ub//ge8WSLf/rhf/3kdbW1mxsbP//mf///yH5BAAAAAAALAAAAAAQAA4AAARe8L1Ekyky67QZ1hLnjM5UUde0ECwLJoExKcppV0aCcGCmTIHEIUEqjgaORCMxIC6e0CcguWw6aFjsVMkkIr7g77ZKPJjPZqIyd7sJAgVGoEGv2xsBxqNgYPj/gAwXEQA7",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <img
          className="favicon"
          src="data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t/f3//Ub//ge8WSLf/rhf/3kdbW1mxsbP//mf///yH5BAAAAAAALAAAAAAQAA4AAARe8L1Ekyky67QZ1hLnjM5UUde0ECwLJoExKcppV0aCcGCmTIHEIUEqjgaORCMxIC6e0CcguWw6aFjsVMkkIr7g77ZKPJjPZqIyd7sJAgVGoEGv2xsBxqNgYPj/gAwXEQA7"
/>,
        "_debugID": 2,
        "_renderedOutput": <img
          className="favicon"
          src="data:image/gif;base64,R0lGODlhEAAOALMAAOazToeHh0tLS/7LZv/0jvb29t/f3//Ub//ge8WSLf/rhf/3kdbW1mxsbP//mf///yH5BAAAAAAALAAAAAAQAA4AAARe8L1Ekyky67QZ1hLnjM5UUde0ECwLJoExKcppV0aCcGCmTIHEIUEqjgaORCMxIC6e0CcguWw6aFjsVMkkIr7g77ZKPJjPZqIyd7sJAgVGoEGv2xsBxqNgYPj/gAwXEQA7"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ImgFromPouch
    attachmentId="favIcon"
    className="favicon"
    doc={
        Object {
            "_attachments": Object {
              "favIcon": Object {
                "content_type": "image/x-icon",
                "digest": "md5-80GKRD59hBCXxxTWnsS8uA==",
                "length": 5430,
                "revpos": 2,
                "stub": true,
              },
              "frozen-page.html": Object {
                "content_type": "text/html;charset=utf-8",
                "digest": "md5-rL1HMBSNfY37snk8/ZBX1w==",
                "length": 549162,
                "revpos": 6,
                "stub": true,
              },
              "screenshot": Object {
                "content_type": "image/png",
                "digest": "md5-iJNPepFWRMhtmqAk/y5eAA==",
                "length": 41222,
                "revpos": 5,
                "stub": true,
              },
            },
            "_id": "page/1499620431844/1337776782",
            "_rev": "6-3b60ef0bcdebea7c54615efb445613a3",
            "extractedMetadata": Object {
              "icon_url": "https://www.google.co.in/images/branding/product/ico/googleg_lodp.ico",
              "provider": "google",
              "title": "Google",
              "url": "https://www.google.co.in/?ei=R2RiWZz1I4OW8QeqwZTIDA",
            },
            "extractedText": Object {
              "bodyInnerText": "Screen-reader users, click here to turn off Google Instant.
        Gmail
        Images
        Sign in
        
        
        India
        Google.co.in offered in: हिन्दी বাংলা తెలుగు मराठी தமிழ் ગુજરાતી ಕನ್ನಡ മലയാളം ਪੰਜਾਬੀ
        Privacy Terms Settings  Use Google.com Advertising Business About
        ",
            },
            "title": "",
            "url": "https://www.google.co.in/?gfe_rd=cr&ei=R2RiWZz1I4OW8QeqwZTIDA",
          }
    }
/>,
}
`;
